src/main/java/mousio/etcd4j/EtcdClient.java

```diff
@@ -276,6 +279,68 @@ public EtcdKeyGetRequest get(String key) {
     return new EtcdKeyGetRequest(client, key, retryHandler);
   }
 
+  /**
+   * Gets the content of the key recursively as a JsonObject
+   * @param path root path (i.e. /path1/path2)
+   * @return JsonNode
+   */
+  public JsonNode getAsJson(String path) throws IOException, EtcdAuthenticationException, TimeoutException, EtcdException {
+    EtcdKeyGetRequest etcdKeyGetRequest = new EtcdKeyGetRequest(client, path, retryHandler).recursive();
+    EtcdKeysResponse dataTree = etcdKeyGetRequest.send().get();
+
+    ObjectNode jNode = JsonNodeFactory.instance.objectNode();
+    dataTree.getNode().getNodes().forEach(node->iterateOverNodes(jNode, node));
+
+    return jNode.at(path);
+  }
+
+  /**
+   * Puts the content of the Json recursively from the specified <i>path</i>
+   * @param path root path (i.e. /path1/path2)
+   * @param data JsonNode
+   */
+  public void putAsJson(String path, JsonNode data) throws IOException, EtcdAuthenticationException, TimeoutException, EtcdException {
+    Map<String, Object> flattened = new JsonFlattener(EtcdUtil.printJson(data))
+            .withFlattenMode(FlattenMode.NORMAL)
+            .withSeparator('/')
+            .flattenAsMap();
+
+    for (Map.Entry<String, Object> entry : flattened.entrySet()) {
+      put(path + "/" + entry.getKey(), String.valueOf(entry.getValue())).send().get();
+    }
+  }
```

Comment:
Not sure if this put/get as json should be part of the api, it may be better to have some utility code to do so

Reply:
Hi, If you consider this should be some utility just let me know so I can refactor the whole thing. It would be great for us to include this feature, doesn't matter if it goes on a separate utility class.

Reply:
I would say it should go in EtcdUtil

Reply:
All right, working on it! I'll close this PR